js/src/components/tree-select-control/options.js

```diff
@@ -81,17 +85,53 @@ const Options = ( {
 	 * @param {Option} option The option to be expanded or collapsed
 	 */
 	const toggleExpanded = ( option ) => {
+		if ( ! option.children?.length ) return;
+
 		onNodesExpandedChange(
 			nodesExpanded.includes( option.value )
 				? nodesExpanded.filter( ( el ) => option.value !== el )
 				: [ ...nodesExpanded, option.value ]
 		);
 	};
 
-	return options.map( ( option ) => {
+	/**
+	 * Expands the option
+	 *
+	 * @param {Option} option The option to expand
+	 */
+	const expand = ( option ) => {
+		if (
+			! option.children?.length ||
+			nodesExpanded.includes( option.value )
+		)
+			return;
+		onNodesExpandedChange( [ ...nodesExpanded, option.value ] );
+	};
+
+	/**
+	 * Alters the node with some keys for accessibility
+	 * ArrowRight - Expands the node
+	 * ArrowLeft - Collapses the node
+	 * Space - Checks/Unchecks the node
+	 *
+	 * @param {Event} event The KeyDown event
+	 * @param {Option} option The option where the event happened
+	 * @param {boolean} isExpanded True if the node is expanded, false otherwise
+	 */
+	const handleKeyDown = ( event, option, isExpanded ) => {
+		if ( event.key === ARROW_RIGHT && ! isExpanded ) {
+			toggleExpanded( option );
+		} else if ( event.key === ARROW_LEFT && isExpanded ) {
+			toggleExpanded( option );
+		} else if ( event.key === SPACE ) {
+			onChange( event.target.checked, option );
```

Comment:
💅  or ❓ 
I just noticed that when pressing Space key on a focused checkbox, the default behavior is already the toggle of checked and unchecked states. And I tested Chrome, FireFox, and Safari on macOS, they can work correctly after removing this.

And removing it could also make the `onChange` callback to be triggered correctly because the `event.target.checked` state got from here is the opposite. The final result of state is still correct due to the following `onChange` event is triggered after it.

https://github.com/woocommerce/google-listings-and-ads/blob/db75784d1b83d67a1d38367fffe2c89373cd898b/js/src/components/tree-select-control/options.js#L178-L180

Reply:
Thanks @eason9487  - To be implemented in next PR within this feature https://github.com/woocommerce/google-listings-and-ads/pull/1336#discussion_r829755331